src/pages/home/report/ReportActionItem.js

```diff
@@ -95,105 +96,89 @@ const propTypes = {
 
 const defaultProps = {
     draftMessage: '',
-    hasOutstandingIOU: false,
     preferredSkinTone: CONST.EMOJI_DEFAULT_SKIN_TONE,
     personalDetails: {},
     shouldShowSubscriptAvatar: false,
+    hasOutstandingIOU: false,
     betas: [],
 };
 
-class ReportActionItem extends Component {
-    constructor(props) {
-        super(props);
-        this.popoverAnchor = undefined;
-        this.state = {
-            isContextMenuActive: ReportActionContextMenu.isActiveReportAction(props.action.reportActionID),
-        };
-        this.checkIfContextMenuActive = this.checkIfContextMenuActive.bind(this);
-        this.showPopover = this.showPopover.bind(this);
-        this.renderItemContent = this.renderItemContent.bind(this);
-        this.toggleReaction = this.toggleReaction.bind(this);
-    }
-
-    shouldComponentUpdate(nextProps, nextState) {
-        return (
-            this.props.displayAsGroup !== nextProps.displayAsGroup ||
-            this.props.draftMessage !== nextProps.draftMessage ||
-            this.props.isMostRecentIOUReportAction !== nextProps.isMostRecentIOUReportAction ||
-            this.props.hasOutstandingIOU !== nextProps.hasOutstandingIOU ||
-            this.props.shouldDisplayNewMarker !== nextProps.shouldDisplayNewMarker ||
-            !_.isEqual(this.props.action, nextProps.action) ||
-            this.state.isContextMenuActive !== nextState.isContextMenuActive ||
-            lodashGet(this.props.report, 'statusNum') !== lodashGet(nextProps.report, 'statusNum') ||
-            lodashGet(this.props.report, 'stateNum') !== lodashGet(nextProps.report, 'stateNum') ||
-            this.props.translate !== nextProps.translate
-        );
-    }
+function ReportActionItem(props) {
+    const [isContextMenuActive, setIsContextMenuActive] = useState(ReportActionContextMenu.isActiveReportAction(props.action.reportActionID));
+    const textInputRef = useRef();
+    const popoverAnchorRef = useRef();
 
-    componentDidUpdate(prevProps) {
-        if (prevProps.draftMessage || !this.props.draftMessage) {
```

Comment:
Coming from https://github.com/Expensify/App/issues/23886

Sometimes, the component will re-render when it is out of focus or the page reloaded, making the input focus unnecessary. To prevent this, we should keep the check against the previous `draftMessage`, and only focus the input if the `draftMessage` is changed.